onnxruntime/core/providers/webnn/builders/impl/attention_op_builder.cc

```diff
@@ -0,0 +1,568 @@
+    true_present_value = model_builder.GetBuilder().call<emscripten::val>(
+        "reshape", true_present_value, emscripten::val::array(group_broadcast_tensor_shape_3), common_options);
+  } else {  // no need for broadcast
+    true_present_key = present_key;
+    true_present_value = present_value;
+  }
+
+  // true_present_key(B,N,P,H) -> transpose(B,N,H,P)
+  options.set("permutation", emscripten::val::array(std::vector<uint32_t>({0, 1, 3, 2})));
+  options.set("label", node.Name() + "/GQA/present_key/transpose");
+  true_present_key = model_builder.GetBuilder().call<emscripten::val>("transpose", true_present_key, options);
+
+  common_options.set("label", node.Name() + "/GQA/qkv/matmul_1");
+  emscripten::val matmul_output =
+      model_builder.GetBuilder().call<emscripten::val>("matmul", new_query, true_present_key, common_options);
+
+  std::vector<float> scale({static_cast<float>(sqrt(head_size))});
+  emscripten::val desc_scale = emscripten::val::object();
+  ORT_RETURN_IF_NOT(SetWebnnDataType(desc_scale, ONNX_NAMESPACE::TensorProto_DataType_FLOAT), "Unsupported data type");
+  emscripten::val dims_scale = emscripten::val::array(std::vector<uint32_t>({1}));
+  desc_scale.set("dimensions", dims_scale);
+  desc_scale.set("shape", dims_scale);
+  emscripten::val scale_buffer = emscripten::val::global("Float32Array").new_(emscripten::val::array(scale));
+  emscripten::val scale_constant =
+      model_builder.GetBuilder().call<emscripten::val>("constant", desc_scale, scale_buffer);
+
+  common_options.set("label", node.Name() + "/GQA/qkv/div");
+  emscripten::val div_output =
+      model_builder.GetBuilder().call<emscripten::val>("div", matmul_output, scale_constant, common_options);
+
+  // static_cast<int64_t>(batch_size), static_cast<int64_t>(num_heads), static_cast<int64_t>(qkv_sequence_length),
+  // static_cast<int64_t>(past_sequence_length)
+  std::vector<int32_t> mask_shape_ones_shape(batch_size * num_heads * qkv_sequence_length * past_sequence_length, 1);
+  emscripten::val desc_mask_shape_ones_shape = emscripten::val::object();
+  ORT_RETURN_IF_NOT(SetWebnnDataType(desc_mask_shape_ones_shape, ONNX_NAMESPACE::TensorProto_DataType_INT32),
+                    "Unsupported data type");
+  emscripten::val dims_mask_shape =
+      emscripten::val::array(std::vector<uint32_t>({batch_size, num_heads, qkv_sequence_length, past_sequence_length}));
+  desc_mask_shape_ones_shape.set("dimensions", dims_mask_shape);
+  desc_mask_shape_ones_shape.set("shape", dims_mask_shape);
+  emscripten::val mask_shape_ones_shape_buffer =
+      emscripten::val::global("Int32Array").new_(emscripten::val::array(mask_shape_ones_shape));
+  emscripten::val mask_shape_ones_shape_constant = model_builder.GetBuilder().call<emscripten::val>(
+      "constant", desc_mask_shape_ones_shape, mask_shape_ones_shape_buffer);
+
+  options.set("label", node.Name() + "range_of_mask_shape");
```

Comment:
At this time, this `options` contains `permutation` member, it's better to remove this member here or create a new `options`.